blocks/store/selectors.js

```diff
@@ -25,9 +27,32 @@ export const getBlockTypes = createSelector(
  *
  * @return {Object?} Block Type.
  */
-export function getBlockType( state, name ) {
-	return state.blockTypes[ name ];
-}
+export const getBlockType = createSelector(
+	( state, name ) => {
+		const blockTypeDefinition = state.blockTypes[ name ];
+		const blockTypeImplementation = state.implementations[ name ];
+
+		if ( ! blockTypeDefinition || ! blockTypeImplementation ) {
+			return null;
+		}
+
+		return {
+			...blockTypeDefinition,
+			...blockTypeImplementation,
+			attributes: mapValues( blockTypeDefinition.attributes, ( attribute, key ) => {
+				const implementationAttribute = blockTypeImplementation.attributes ? blockTypeImplementation.attributes[ key ] : {};
```

Comment:
Using `get( blockTypeImplementation.attributes, [ key ], {} )` here might be easier to understand.